data/data.go

```diff
@@ -139,35 +136,80 @@ func NewestSemVer(v1 string, v2 string) (string, error) {
 	return v1, nil
 }
 
-// getRCItems is a helper function that returns a slice of
-// ReplicationController objects given a rc.Lister interface
-func getRCItems(rcLister rc.Lister) ([]api.ReplicationController, error) {
-	rcs, err := rcLister.List(api.ListOptions{
-		LabelSelector: labels.Everything(),
-	})
-	if err != nil {
-		return []api.ReplicationController{}, err
-	}
-	return rcs.Items, nil
-}
-
 // GetDoctorInfo collects doctor info and return DoctorInfo struct
 func GetDoctorInfo(
-	c InstalledData,
+	c InstalledData, // workflow cluster data
+	k k8s.RunningK8sData, // k8s data
 	i ClusterID,
 	v AvailableComponentVersion,
-	secretGetterCreator KubeSecretGetterCreator,
 ) (models.DoctorInfo, error) {
-	cluster, err := GetCluster(c, i, v, secretGetterCreator)
+	cluster, err := GetCluster(c, i, v)
 	if err != nil {
-		return models.DoctorInfo{}, err
+		log.Println("unable to get workflow cluster data")
 	}
+	nodes := getK8sNodes(k)
+	var namespaces []*models.Namespace
+	namespaces = append(namespaces, getK8sDeisNamespace(k))
```

Comment:
replace the above two lines with: `namespaces := []*models.Namespace{getK8sDeisNamespace(k)}`